WordPress/wporg-main-2022
# Navigation Block: Add option to show a hardcoded "dynamic menu"

See #266 — Core navigation menus work by using a `ref`, an ID to a menu post. On the main site, this is fine, but on rosetta sites, this does not work because that post is not available on the local site.

For example, the About subpage should have a right-hand menu:

| English | Rosetta |
|---|---|
| ![Screen Shot 2023-07-20 at 15 34 51](https://github.com/WordPress/wporg-main-2022/assets/541093/6c55de10-a32b-4123-9d02-2fb5032b021c) | ![Screen Shot 2023-07-20 at 15 34 44](https://github.com/WordPress/wporg-main-2022/assets/541093/e4d5ee74-0602-4a64-8883-a27eb8500665) |

We could do some switch_to_blog work, but that would still load the English menu & links to the English site. Instead, I've added a new attribute to the Navigation block, `menuSlug`. When this is present (and not empty), the menu's content is swapped out for a hardcoded menu which uses the local site's base URL and wraps the menu text in translation wrappers.

I also added some styles for the vertical nav menu format, to match the list of links style, so we can use that on the About page (otherwise the links to the main English site were hardcoded).

### Screenshots

Each menu in these screenshots are dynamically generated, linking to the page on the rosetta site.

| Label | Screenshot |
|---|---|
| About | ![Screen Shot 2023-07-20 at 15 45 45](https://github.com/WordPress/wporg-main-2022/assets/541093/2a5cb028-a5d8-47b5-b04e-4c5462b11096) |
| About subpage | ![Screen Shot 2023-07-20 at 15 42 59](https://github.com/WordPress/wporg-main-2022/assets/541093/fe3d9444-9b6b-47e2-ac7a-61ca9992fefb) |
| Download | ![Screen Shot 2023-07-20 at 15 43 14](https://github.com/WordPress/wporg-main-2022/assets/541093/ec741295-cc09-486a-9823-bf030828646b) |
| Download subpage | ![Screen Shot 2023-07-20 at 15 43 25](https://github.com/WordPress/wporg-main-2022/assets/541093/38501344-3a76-494a-9058-05ea3028e337) |

The dynamic menu doesn't have an in-editor preview, but the menu slug can be controlled in the editor — this at least prevents any regressions when the content is updated.

![nav-in-editor](https://github.com/WordPress/wporg-main-2022/assets/541093/1474ce5c-8756-4253-b80c-94025d67a144)

### How to test the changes in this Pull Request:

1. About, Download, and the subpages should have visible menus regardless of whether those IDs exist on your local site.
2. Create or edit a page with a Navigation block on it
3. Switch it to use a dynamic menu
4. You should be able to control all the other settings (ex, layout & styles)
5. It should render that dynamic menu on the front end, with the settings applied
